lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -126,6 +200,53 @@ getSlicedOperands(OpBuilder &builder, Location loc, ValueRange localIvs,
   return slicedOperands;
 }
 
+// Returns true if linalg op is using its second operand in VNNI format
+static bool isLinalgOpVNNI(linalg::LinalgOp linalgOp) {
```

Comment:
Let's split this method a bit. It is difficult to follow.

Reply:
If I get rid of the pack op check, would you recommend refactoring still?

Reply:
line 211 till 215 are still difficult to follow for me. Maybe moving the operand into a variable would help. Something like:

```
auto firstOperand = linalgOp->getOperands()[1].getType().cast<ShapedType>();
if (linalgOp.getNumLoops() == 7 && linalgOp->getOperands().size() >= 3 && firstOperand.isBF16() && firstOperand.getShape()[firstOperand.getRank() - 1] == 2) {

}
```
 Also this if seems to check two different things: 1 structural property of the ops (i.e., number of loops) and 2 a specific operand of the op. We could split the if by checking first the structural properties and then the operands.